File: AndroidTapToProgress/app/src/main/java/com/example/shakil/androidtaptoprogress/CircleAnimation.java
package com.example.shakil.androidtaptoprogress;

import android.view.animation.Animation;
import android.view.animation.Transformation;

public class CircleAnimation extends Animation {
    Circle circle;
    float angle;
    float endAngle;

    public CircleAnimation(Circle circle, float endAngle) {
        this.circle = circle;
        this.angle = circle.getAngle();
        this.endAngle = endAngle;
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        this.circle.setAngle(this.angle + ((this.endAngle - this.angle) * interpolatedTime));
        this.circle.requestLayout();

    }
}

File: TestKotlinTapToProgress/app/src/main/java/com/example/shakil/testkotlintaptoprogress/CircleAnimation.kt
package com.example.shakil.testkotlintaptoprogress

import android.view.animation.Animation
import android.view.animation.Transformation

class CircleAnimation(private val circle: Circle, private val endangle: Float) : Animation() {
    private val angle: Float

    init {
        this.angle = circle.angle
    }


    override fun applyTransformation(f: Float, transformation: Transformation) {
        this.circle.angle = this.angle + (this.endangle - this.angle) * f
        this.circle.requestLayout()
    }
}
